### ISSUE TITLE
Dictionary suggestions are not working properly in avro phonetic layout. Distro- linux mint 19 cinnamon   

### ISSUE THREAD

**Issue by kuntalkundu** [OpenBangla/OpenBangla-Keyboard#73]
While using avro phonetic layout the drop down suggestion menu is displaying only the current word which has been just written, there are no any other suggestions of probable words are displaying. Suppose if I write 'soumen' the dropdown menu will show সউমেন but what we actually need is সৌমেন as first suggestion and rest are later. I did everything in openbangla settings to solve it but result is same.

**Comment by mominul** [comment]
Actually, you are typing wrong. You should type `sOUmen` because according to Avro Phonetic layout you would need to type `OU` to get `ৌ` kar and it is also a case sensitive.

**Comment by kuntalkundu** [comment]
it's just an example. yes, to write সৌমেন it's important to care the case sensitive letters while typing it with Avro phonetic, but I want to tell about the probable words suggestions which are being shown in Avro suggestion list. Like ঐক্য if I type "oiky" Ibus avro will show me atleast 3 suggestions in drop down menu including ঐক্য but openbangla phonetic only shows me just অইক্য. I'm attaching an example -[https://mega.nz/#!jSAhlapI!h-Hqin4aBIt9vB9ujkMUJgbCPVrDIWLePYTae35pUW0](url)

**Comment by mominul** [comment]
Can you give the screenshots when writing `oiky` with ibus-avro and openbangla? I want to see the difference.

**Comment by kuntalkundu** [comment]
Ibus Avro by Mateen- oiky https://mega.nz/#!ObZFGKTK!mBKkvoZM8mc7i6uTWIdLcBrwc5WBooClFYiweuEHb3w
openBangla- oiky https://mega.nz/#!GKQVjYBJ!bWKPaAuWB19Ta01P68dxKvLsP1Ig1X_j0zxcpd_PG28

**Comment by mominul** [comment]
Seems fine to me :thinking: 

![oiky](https://user-images.githubusercontent.com/9459891/46660189-52f59200-cbd8-11e8-9f85-162c868ea466.png)

Can you try with a different editor eg. gedit?

**Comment by kuntalkundu** [comment]
Mint is come with xed, fork of gedit, is this issue related to text editor? I'm confused! Ok I shall test it too asap. What is the result on LibreOffice 6? I too can remember when I did install ubuntu 18.04 openbangla phonetic works fine at that time. But this issue I have noticed since Manjaro Kde and now on mint 19 which is built on ubuntu 18.04!

**Comment by kuntalkundu** [comment]
yeah, I just tested openBangla keyboard in gedit installed on linuxmint and as expected result remained same. I also installed ubuntu 18.04 on a VM and install OpenBangla there and now it's phonetic suggestions are working fine! I think it's a bug of OBK. Ubuntu came with ibus IM as it's default input tool but other distros like mint or manjaro never come with ibus as default. OBK is an ibus based app and it works great with ubuntu but when ibus being set up manually in other distros result became slightly different (phonetic suggestions not working).

**Comment by kuntalkundu** [comment]
in between time I have tested OBK in Ubbuntu MATE 18.04 too. OBK phonetic suggestions are working correctly also there. So I can guess that OBK shall run much better on Ubuntu or on it's official derivatives. What about Manjaro or Arch version of OBK? I think this is a bug and wish soon it would be fixed for rest of distros based on debian or ubuntu.

**Comment by mominul** [comment]
Can you uninstall and then install OBK again please?

**Comment by kuntalkundu** [comment]
What is the lates version of Obk? I have installed 1.5.0 already. Is there any new release? I just copied and paste the command lines provided in obk website for auto install. Should I use deb file?

**Comment by mominul** [comment]
Oh, no worries then. First uninstall OBK and then install it using by pasting the command line provided in the website. Can you also paste here the terminal output of the installation command?

**Comment by kuntalkundu** [comment]
how can I do that? little bit knowledge about that. I already finished the installation. this time I used .deb file for 18.04. now checking...

**Comment by kuntalkundu** [comment]
after reinstall obk the result remains same https://www.dropbox.com/s/aej62npbuz9x35m/%E0%A6%A6%E0%A7%87%E0%A6%B8.png?dl=0 now I am checking the 16.04 .deb package..

**Comment by mominul** [comment]
Oh thanks! I think I have gotten the root of the cause. Somehow OBK is being unable to load the database.

**Comment by kuntalkundu** [comment]
oh... that's great!! I suspect something like that somehow obk can't get it's database. here I too test the 16.04 deb package in mint19, result is same. Now how could we get the solution?

**Comment by mominul** [comment]
can you follow these instructions, please?
At first, run this command in your terminal:
```
$ /usr/share/openbangla-keyboard/OpenBangla-Engine
```
After then open a text editor and press some keys. Afterward press <kbd>Ctrl</kbd> + <kbd>C</kbd> in the terminal to exit that command. There should be some debug information printed to your terminal, copy them and paste it here.

Thanks!

**Comment by kuntalkundu** [comment]
wait... I am doing

**Comment by kuntalkundu** [comment]
k@k-VirtualBox:~$ /usr/share/openbangla-keyboard/OpenBangla-Engine
[OpenBangla-Error]: [AutoCorrect]: Error: Couldn't open user specific AutoCorrect file!
QIODevice::read (QFile, "/home/k/.OpenBangla-Keyboard/autocorrect.json"): device not open
QSqlDatabase: QSQLITE driver not loaded
QSqlDatabase: available drivers: 
QSqlDatabase: an instance of QCoreApplication is required for loading driver plugins
[OpenBangla-Error]: [CacheManager:Load]: Error couldn't open save file.
[OpenBangla-Info]: [IM:iBus]: Creating IM Engine
[OpenBangla-Info]: [IM:iBus]: IM enabled
[OpenBangla-Error]: [AutoCorrect]: Error: Couldn't open user specific AutoCorrect file!
QIODevice::read (QFile, "/home/k/.OpenBangla-Keyboard/autocorrect.json"): device not open
^C
k@k-VirtualBox:~$

**Comment by kuntalkundu** [comment]
oh no... why the lines looks like this!! strike through!

**Comment by kuntalkundu** [comment]
I have uninstalled the 16.04 version and reinstall the latest through terminal commands. Here the terminal output-
k@k-VirtualBox:~$ bash -c "$(wget -q https://raw.githubusercontent.com/OpenBangla/OpenBangla-Keyboard/master/tools/install.sh -O -)"
|-------------------------------------------------------|
|-----Welcome to the OpenBangla Keyboard installer------|
|-------------------------------------------------------|

This helper script will install/update OpenBangla Keyboard in your system.
You might need to type your password for the installation to proceed.

After the installation completes successfully, you would need to configure
your desktop environment for OpenBangla Keyboard to work.
Visit https://github.com/OpenBangla/OpenBangla-Keyboard/wiki/Configuring-Environment
for instructions on how to configure your desktop environment.
--------------------------------------------------------

cat: /etc/upstream-release: Is a directory
Downloading installation package for ubuntu ...
/home/k/OpenBangla. 100%[===================>]   5.29M   768KB/s    in 8.6s    
[sudo] password for k: 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Note, selecting 'openbangla-keyboard' instead of '/home/k/OpenBangla.deb'
The following NEW packages will be installed:
  openbangla-keyboard
0 upgraded, 1 newly installed, 0 to remove and 5 not upgraded.
Need to get 0 B/5,551 kB of archives.
After this operation, 15.7 MB of additional disk space will be used.
Get:1 /home/k/OpenBangla.deb openbangla-keyboard amd64 1.5.0 [5,551 kB]
Selecting previously unselected package openbangla-keyboard.
(Reading database ... 278216 files and directories currently installed.)
Preparing to unpack /home/k/OpenBangla.deb ...
Unpacking openbangla-keyboard (1.5.0) ...
Processing triggers for mime-support (3.60ubuntu1) ...
Processing triggers for desktop-file-utils (0.23+linuxmint3) ...
Setting up openbangla-keyboard (1.5.0) ...
Processing triggers for gnome-menus (3.13.3-11ubuntu1.1) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
k@k-VirtualBox:~$

**Comment by mominul** [comment]
This is a missing dependency error. Can you install `qt5-default` package? You can install it like this:
```
sudo apt install qt5-default
```

Then can you again try running `/usr/share/openbangla-keyboard/OpenBangla-Engine`?

**Comment by kuntalkundu** [comment]
sure, wait few moment

**Comment by kuntalkundu** [comment]
/usr/share/openbangla-keyboard/OpenBangla-Engine
[OpenBangla-Error]: [AutoCorrect]: Error: Couldn't open user specific AutoCorrect file!
QIODevice::read (QFile, "/home/k/.OpenBangla-Keyboard/autocorrect.json"): device not open
QSqlDatabase: QSQLITE driver not loaded
QSqlDatabase: available drivers: 
QSqlDatabase: an instance of QCoreApplication is required for loading driver plugins
[OpenBangla-Error]: [CacheManager:Load]: Error couldn't open save file.
[OpenBangla-Info]: [IM:iBus]: Creating IM Engine
[OpenBangla-Info]: [IM:iBus]: IM enabled
[OpenBangla-Error]: [AutoCorrect]: Error: Couldn't open user specific AutoCorrect file!
QIODevice::read (QFile, "/home/k/.OpenBangla-Keyboard/autocorrect.json"): device not open
^C

**Comment by kuntalkundu** [comment]
thanks for the promise <3 yes this is very irritating. Please check the AUR package too, I am not using arch based distro now, so can not tell you the updates on this bug but last time I had installed Manjaro KDE and what I can remember the bug was also there most probably.

**Comment by mominul** [comment]
I have committed a fix in the master branch. If you want to try it, I can give you the compiling instructions.

**Comment by shakib609** [comment]
> thanks for the promise <3 yes this is very irritating. Please check the AUR package too, I am not using arch based distro now, so can not tell you the updates on this bug but last time I had installed Manjaro KDE and what I can remember the bug was also there most probably.

I am not facing any issues in Manjaro KDE.

**Comment by kuntalkundu** [comment]
Could you provide a screenshot of Obk phonetic suggestions in manjaro kde?

**Comment by mominul** [comment]
@shakib609 @kuntalkundu This bug is some kind of missing dependency error. As @shakib609 has said it shouldn't happen in most scenarios but for some unknown reason QT seems unable to load the SQLite library sometimes. The fix I have pushed completely removes the use of SQLite database, so there is no need for QT to load SQLite library anymore. Thus this issue can be called fixed. :tada: 

I am thinking to make a new release with the fix included.

**Comment by kuntalkundu** [comment]
that's great... countdown for 1.5.1 now starts 💃

**Comment by mominul** [comment]
Released version 1.5.1 with the fix! :tada:

**Comment by kuntalkundu** [comment]
অনেক ধন্যবাদ মমিনুল ভাই এবং ওপেন বাংলা টিমকে বিষয়টি গুরুত্ব সহকারে বিবেচনা করার জন্য এবং সমাধানের জন্য। ওপেনবাংলার আরও প্রসার হোক, মাতৃভাষায় মত প্রকাশ আরও বৃদ্ধি পাক এই কামনা করি।
